### PR TITLE
Feat: Filter quiz sessions based on batch ID

### PR DESCRIPTION
**Pull request by Bahugunajii**
### ADR
https://www.notion.so/avantifellows/ADR-Class-Batch-Metadata-for-Quiz-Sessions-cd84df30f7db46ab91bf023ea6aa9e68

### Description
- Filter quiz sessions based on the `batch_id` in metadata

### Checklist
- [x] Self-testing

### CODE REVIEW

File: lib/dbservice_web/controllers/user_controller.ex [hunk 1]
```diff
@@ -201,14 +201,42 @@ defmodule DbserviceWeb.UserController do
   defp get_group(group, quiz_flag) do
     child_id = group.child_id
     batch = Batches.get_batch!(child_id)
+    class_batch_id = batch.batch_id
     quiz_id = batch.parent_id
```
Comment (suryabulusu) [inline lib/dbservice_web/controllers/user_controller.ex:205]: quiz_batch _id

File: lib/dbservice_web/controllers/user_controller.ex [hunk 2]
```diff
@@ -201,14 +201,42 @@ defmodule DbserviceWeb.UserController do
...
     quiz_group = Groups.get_group_by_child_id(quiz_id)
```
Comment (suryabulusu) [inline lib/dbservice_web/controllers/user_controller.ex:206]: quiz_batch_group

File: lib/dbservice_web/controllers/user_controller.ex [hunk 3]
```diff
@@ -201,14 +201,42 @@ defmodule DbserviceWeb.UserController do
...
+
+      _ ->
+        if quiz_flag do
+          if !is_nil(session.meta_data["batch_id"]) do
+            session.meta_data["batch_id"] == class_batch_id and session.platform == "quiz"
+          else
+            false
+          end
+        else
+          session.platform == "meet"
```
Comment (suryabulusu) [inline lib/dbservice_web/controllers/user_controller.ex:237]: why this? you can just put true maybe? if quiz flag is false then eveyr other platform is allowed. not just meet

**Review by suryabulusu — APPROVED** [review]
left comments

File: lib/dbservice_web/controllers/user_controller.ex (hunk 3 above)
Comment (Bahugunajii) [reply lib/dbservice_web/controllers/user_controller.ex:237]: But for live classes, shouldn't the platform be always meet?

Comment (Bahugunajii) [reply lib/dbservice_web/controllers/user_controller.ex:237]: In Gurukul, we show only Live classes and Quiz. My intent was to show "meet" sessions for live classes and "quiz" sessions for Tests.

File: lib/dbservice_web/controllers/user_controller.ex (hunk 1 above)
Comment (Bahugunajii) [reply lib/dbservice_web/controllers/user_controller.ex:203]: But quiz_id is PK here

**Pull request merged**